synapse/rest/client/v1/room.py

```diff
@@ -229,60 +226,22 @@ def on_POST(self, request, room_identifier, txn_id=None):
             allow_guest=True,
         )
 
-        # the identifier could be a room alias or a room id. Try one then the
-        # other if it fails to parse, without swallowing other valid
-        # SynapseErrors.
+        handler = self.handlers.room_member_handler
 
         identifier = None
-        is_room_alias = False
+        hosts = []
         try:
             identifier = RoomAlias.from_string(room_identifier)
-            is_room_alias = True
-        except SynapseError:
-            identifier = RoomID.from_string(room_identifier)
+            identifier, hosts = yield handler.lookup_room_alias(identifier)
```

Comment:
I wonder if ``identifier.startswith('#')`` would be clearer than relying on the exceptions thrown.